Formula/cppinsights.rb

```diff
@@ -0,0 +1,32 @@
+class Cppinsights < Formula
+  desc "See your source code with the eyes of a compiler"
+  homepage "https://cppinsights.io/"
+  url "https://github.com/andreasfertig/cppinsights/archive/refs/tags/v_0.9.tar.gz"
+  sha256 "cebb6a062677ee3975ff757e4300a17f42e3fab6da02ad01f3f141fb91f09301"
+  license "MIT"
+
+  depends_on "cmake" => :build
+
+  depends_on "llvm@15"
+  depends_on macos: :monterey
+
+  fails_with :clang do
+    build 1300
+    cause "Requires C++20"
+  end
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```

Comment:
```suggestion
  def install
    ENV["HOMEBREW_CC"] = Formula["llvm@15"].opt_bin/"clang" if DevelopmentTools.clang_build_version <= 1300
    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
```
This might help you get rid of the Monterey requirement.